From 04e49e980b5a64df4bbfc40616f8df86491db784 Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Sun, 13 Mar 2022 08:56:50 -0400
Subject: [PATCH] Fix exporter-{jaeger,otlp,zipkin} install paths
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

These “convenience” packages were incorrectly installed in
“site-packages/src”, although the actual implementation packages they
depended on were installed in the correct paths (which is probably why
this was not noticed sooner).

Fixes #2020.
---
 CHANGELOG.md                                     | 4 ++++
 exporter/opentelemetry-exporter-jaeger/setup.cfg | 6 +++++-
 exporter/opentelemetry-exporter-otlp/setup.cfg   | 5 +++++
 exporter/opentelemetry-exporter-zipkin/setup.cfg | 5 +++++
 4 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG.md b/CHANGELOG.md
index 17966ed52c..b31a3b58d8 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased](https://github.com/open-telemetry/opentelemetry-python/compare/v1.10.0-0.29b0...HEAD)
 
+- Fix incorrect installation of some exporter “convenience” packages into
+  “site-packages/src”
+  ([#2525](https://github.com/open-telemetry/opentelemetry-python/pull/2525))
+
 ## [1.10.0-0.29b0](https://github.com/open-telemetry/opentelemetry-python/releases/tag/v1.10.0-0.29b0) - 2022-03-10
 
 
diff --git a/exporter/opentelemetry-exporter-jaeger/setup.cfg b/exporter/opentelemetry-exporter-jaeger/setup.cfg
index d16e075227..0c3fe8a311 100644
--- a/exporter/opentelemetry-exporter-jaeger/setup.cfg
+++ b/exporter/opentelemetry-exporter-jaeger/setup.cfg
@@ -37,12 +37,16 @@ classifiers =
 
 [options]
 python_requires = >=3.6
-
+package_dir=
+    =src
 packages=find_namespace:
 install_requires =
     opentelemetry-exporter-jaeger-proto-grpc == 1.10.0
     opentelemetry-exporter-jaeger-thrift == 1.10.0
 
+[options.packages.find]
+where = src
+
 [options.extras_require]
 test =
 
diff --git a/exporter/opentelemetry-exporter-otlp/setup.cfg b/exporter/opentelemetry-exporter-otlp/setup.cfg
index 9e6a1e3f4a..6dfc63ee52 100644
--- a/exporter/opentelemetry-exporter-otlp/setup.cfg
+++ b/exporter/opentelemetry-exporter-otlp/setup.cfg
@@ -37,11 +37,16 @@ classifiers =
 
 [options]
 python_requires = >=3.6
+package_dir=
+    =src
 packages=find_namespace:
 install_requires =
     opentelemetry-exporter-otlp-proto-grpc == 1.10.0
     opentelemetry-exporter-otlp-proto-http == 1.10.0
 
+[options.packages.find]
+where = src
+
 [options.entry_points]
 opentelemetry_traces_exporter =
     otlp = opentelemetry.exporter.otlp.proto.grpc.trace_exporter:OTLPSpanExporter
diff --git a/exporter/opentelemetry-exporter-zipkin/setup.cfg b/exporter/opentelemetry-exporter-zipkin/setup.cfg
index 910c7ccd80..c431d302dd 100644
--- a/exporter/opentelemetry-exporter-zipkin/setup.cfg
+++ b/exporter/opentelemetry-exporter-zipkin/setup.cfg
@@ -37,11 +37,16 @@ classifiers =
 
 [options]
 python_requires = >=3.6
+package_dir=
+    =src
 packages=find_namespace:
 install_requires =
     opentelemetry-exporter-zipkin-json == 1.10.0
     opentelemetry-exporter-zipkin-proto-http == 1.10.0
 
+[options.packages.find]
+where = src
+
 [options.extras_require]
 test =
 
